src/Smart.Design.Razor/TagHelpers/TableOfContent/TableOfContentHtmlGenerator.cs

```diff
@@ -0,0 +1,41 @@
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Mvc.Rendering;
+
+namespace Smart.Design.Razor.TagHelpers.TableOfContent;
+
+/// <summary>
+/// Create the grid row  in a table header.
+/// To use in conjunction with <see cref="GridListTagHelper"/>
+/// </summary>
+public class TableOfContentHtmlGenerator : ITableOfContentHtmlGenerator
+{
+    public TagBuilder GenerateListOfItems(List<string> titles)
+    {
+        var list = new TagBuilder("ol");
+        list.AddCssClass("c-table-of-content__nav relativeWidth stickyPosition");
+
+        int i = 1;
+        foreach (var title in titles)
+        {
+            var li = new TagBuilder("li");
+
+            var link = new TagBuilder("a");
+            link.AddCssClass("c-table-of-content__nav__item");
+            link.Attributes["href"] = $"#section_{i}";
+            link.InnerHtml.Append(title);
+
+            li.InnerHtml.AppendHtml(link);
+            list.InnerHtml.AppendHtml(li);
+
+            i++;
+        }
```

Comment:
Not a blocking change for the PR but it would be a nice usecase where using a simple _for_ loop might be a good alternative  since you have an incremented index. It's in theory more performant too but seeing the use case, it's more a matter of preference I guess :b
```suggestion
      for(var i = 0 ; i < titles.Count; i++)
        {
            var li = new TagBuilder("li");
            var link = new TagBuilder("a");
            link.AddCssClass("c-table-of-content__nav__item");
            link.Attributes["href"] = $"#section_{i + 1}";
            link.InnerHtml.Append(titles[i]);

            li.InnerHtml.AppendHtml(link);
            list.InnerHtml.AppendHtml(li);
        }
```